control/controlclient/map.go

```diff
@@ -664,9 +664,22 @@ func peerChangeDiff(was tailcfg.NodeView, n *tailcfg.Node) (_ *tailcfg.PeerChang
 				pc().Cap = n.Cap
 			}
 		case "CapMap":
-			if n.CapMap != nil {
-				pc().CapMap = n.CapMap
+			if len(n.CapMap) != was.CapMap().Len() {
+				if n.CapMap == nil {
+					pc().CapMap = make(tailcfg.NodeCapMap)
+				} else {
+					pc().CapMap = n.CapMap
```

Comment:
looks like all the other return paths returning a diff don't alias memory provided by the caller

(line 678 too)

Reply:
Can I make that a follow on PR to unblock Claire? The existing code was already aliasing memory.

Reply:
k

Reply:
@sailorfrag pointed out that since the value type for NodeCapMap is of type tailcfg.RawMessage, which is a string thus immutable. So - aliasing memory is ok?

Reply:
The RawMessage yes, the map itself, no